src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+﻿using System;
+using System.Reactive.Disposables;
+using System.Windows.Input;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="Button"/> and the secondary part opens a flyout.
+    /// </summary>
+    [PseudoClasses(
+        ":disabled",
+        ":secondaryButtonRight",
+        ":secondaryButtonSpan",
+        ":checkedFlyoutOpen",
+        ":flyoutOpen",
+        ":checkedTouchPressed",
+        ":checked",
+        ":checkedPrimaryPressed",
+        ":checkedPrimaryPointerOver",
+        ":checkedSecondaryPressed",
+        ":checkedSecondaryPointerOver",
+        ":touchPressed",
+        ":primaryPressed",
+        ":primaryPointerOver",
+        ":secondaryPressed",
+        ":secondaryPointerOver")]
+    public class SplitButton : ContentControl, ICommandSource
+    {
+        /// <summary>
+        /// Raised when the user presses the primary part of the <see cref="SplitButton"/>.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> Click
+        {
+            add => AddHandler(ClickEvent, value);
+            remove => RemoveHandler(ClickEvent, value);
+        }
+
+        /// <summary>
+        /// Defines the <see cref="Click"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> ClickEvent =
+            RoutedEvent.Register<SplitButton, RoutedEventArgs>(
+                nameof(Click),
+                RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="Command"/> property.
+        /// </summary>
+        public static readonly DirectProperty<SplitButton, ICommand> CommandProperty =
+            AvaloniaProperty.RegisterDirect<SplitButton, ICommand>(
+                nameof(Command),
+                splitButton => splitButton.Command,
+                (splitButton, command) => splitButton.Command = command,
+                enableDataValidation: true);
```

Comment:
You don't need to register new AvaloniaProperty but use already registered instance from Button.

Reply:
In WPF/Avalonia it's very useful to share same Property/Event definition, when you work with abstract methods like control.GetValue(Button.CommandProperty) - no need to duplicate it for two different CommandProperty.

Reply:
SplitButton intentionally doesn't derive from Button -- it derives from ContentControl. While I may be able to do as you suggest for a StyledProperty I can't think it would work for a DirectProperty? Even so is this a good idea considering they don't inherit?

Reply:
Max just means .AddOwner the property, it's already defined in Button and it works for both Styled and Direct

````
public static readonly DirectProperty<SplitButton, ICommand> CommandProperty = 
Button.CommandProperty.AddOwner<SplitButton>(x => x.Command, (x,v) => x.Command v);

or 
public static readonly StyledProperty<object> CommandParameterProperty = 
Button.CommandParameterProperty.AddOwner<SplitButton>();
````

Reply:
Yes, properties should be reused even if it's not inherited.

Reply:
Alright I can change that. I can see how that might make sense.

Reply:
This should be resolved: I added owners for all SplitButton properties. Note ToggleSplitButton couldn't do that for reasons mentioned separately.